src/services/ClientService.ts

```diff
@@ -328,6 +329,43 @@ class ClientService {
     return ticket;
   }
 
+  static async getServiceRequestSummary(): Promise<ServiceRequestSummary[]> {
+    const { data, error } = await supabaseClient
+      .from('service_requests')
+
+      .select(`
+      id,
+      description,
+      created_at,
+      service_category,
+      clients(first_name),
+      pets(name),
+      team_members(first_name)
+      `)
+      .order('created_at', { ascending: false })
+      .limit(10);
```

Comment:
With this constraint, the pagination will only be able to use 2 pages. Maybe we increase this to like 100? But we will still need server side pagination in order to grow the application.

Reply:
-  Why exactly do you want 100 records shown all at once? when they get around 50-75 service requests per week? 
- Can I ask why you think this pagination is limited to only two pages,  since from your comment it is not a question/wonder that is a statement?
- No it is not limited to two pages only-  with the current limit set, if there are 100 records passed on the table it will be 10 pages holding 10 records on each page. `limit(int)` refers to how many rows we want displayed on each page
- These screenshots were taken on my local machine, I created 7 records manually because **I have to create them manually**... If there is a fast way to create 100s of records in minutes I will do so and if I knew about it I would have done it so but sadly I do not. 
- Maybe we should think about creating testing seed data?
- From the meeting, they said they get 50- 75 requests a week. If they are closing these tickets they will be moving out of this table and on the closet tickets section which we will only be rendering I think Kate said 20 recently closed tickets so if we use this same table still does not make sense to flood their dashboard of 3-4 tables showing 100 of records each.
- Supabase has built pagination on the server side [here](https://www.restack.io/docs/supabase-knowledge-supabase-pagination-guide) if they get 50 to 75  the most weekly it will take a few months for their db to have 1000 records. You can add a few lines of code to the existing queries to leverage the pagination feature built-in.

Reply:
I wrote my answers below so you know my logic, but I think I might not understand. I think you're right, but it would be nice to set up a call for you to explain it to me. I think the best thing would be to set up a database with like 150 records and see how the table handles it

> Why exactly do you want 100 records shown all at once? when they get around 50-75 service requests per week?
I don't think 100 records should be shown at once, just that the SQL query should ask for a higher number, like 100. The frontend table would only display 10 or so at a time
> Can I ask why you think this pagination is limited to only two pages, since from your comment it is not a question/wonder - it is a statement?
My understanding might be wrong, but this was my logic. The SQL query here returns the latest records, but up to a limit of 10. The pages that you are displaying on the table are of size 5. If the query returns at most 10 records and the front end table displays 5 records at a time, then the max number of pages could be 10/5 = 2
> No, it is not limited to two pages only- with the current limit set, if there are 100 records passed on the table it will be 10 pages holding 10 records on each page. limit(int) refers to how many rows we want displayed on each page
Oh I think I must have a misunderstanding then! So will this query go through the whole database using SQL?
> These screenshots were taken on my local machine, I created 7 records manually because I have to create them manually... If there is a fast way to create 100s of records in minutes I will do so and if I knew about it I would have done it so but sadly I do not.
I could help with that! We could use a query with a loop to do that. I could help you write that
> Maybe we should think about creating testing seed data?
Great idea!
>From the meeting, they said they get 50- 75 requests a week. If they are closing these tickets they will be moving out of this table and on the closet tickets section which we will only be rendering I think Kate said 20 recently closed tickets so if we use this same table still does not make sense to flood their dashboard of 3-4 tables showing 100 of records each.
Definitely don't want massive tables, I just want to make sure we can handle a large dataset when we use this tool for different use cases.
> Supabase has built pagination on the server side [here](https://www.restack.io/docs/supabase-knowledge-supabase-pagination-guide) if they get 50 to 75 the most weekly it will take a few months for their db to have 1000 records. You can add a few lines of code to the existing queries to leverage the pagination feature built-in.
Oh great let me check that out

Reply:
@tristan-88 is right we don't expect too many records, but we do need to build a solution that will grow with SH. 
This function is specifically for loading recent tickets on the dashboard. We will have other pages that show the entire client list, or pet list, or service details list. When we get to those pages, we will likely have different fields to show which means writing a different query, at which point we can use limit and offset, as Tristian linked to in the Supabase docs.

I don't think we're looking to have pagination for the Recent Tickets list as it appears on the dashboard. The task, per my understanding, is to build re-usable client components. So the pagination Tristian has on the client will work for this use case. A future table on a "all clients" page will need to implement its query and data fetching to take server pagination into consideration.

Just my opinion. What do you think @DillonOLeary ?

Reply:
Here's a query to generate a ton of rows. You might need to edit the uuids:
```SQL
INSERT INTO service_requests (description, pet_id, client_id, team_member_id, service_category, request_source)
SELECT 'Dummy description', 
       'c943b5cb-ae8d-48ac-af3a-09861dec3f34', 
       'd65f3dac-19cc-4083-81b4-ceb7738e2440', 
       '51c85673-4fbf-4ccf-9cd6-b93bdd2cc39f', 
       'c44b5397-0ba8-4601-9d99-a60fc4ace704', 
       '571832e4-027e-4cc0-a05a-ad3b2d5249ae'
FROM generate_series(1, 100);
```

Reply:
With that data, it looks like only 2 pages were generated:

<img width="832" alt="Screenshot 2024-07-05 at 4 58 29 PM" src="https://github.com/digitalaidseattle/seattle-humane-app/assets/25556678/bb164675-9a2e-4785-9fc8-e813e9c72d7e">

If I change the query to `.limit(50)` I get half the records back:

<img width="855" alt="Screenshot 2024-07-05 at 4 59 14 PM" src="https://github.com/digitalaidseattle/seattle-humane-app/assets/25556678/2c9cbdd6-8a0f-46db-9a76-c482a5e12b9f">

Reply:
Yea, @tristan-88 we need 2 queries. One to get the total number of records (to set on the pagination component) and another to get the data rows themselves.

Also, how about we add env variables to make it easy to configure the number of records shown in recent tickets? Later we should also have env vars that control the limit size for server pagination so we can change them without deploying code changes. 

```
NEXT_PUBLIC_RECENT_TICKETS_DISPLAYED = 10
# And in the future for the server we could use something like:
# QUERY_PAGE_SIZE = 
```

Reply:
@sharrief I think you are right, the only thing that would be important is that the user knows we are not returning all the data. In the screenshot above it says `Showing 1 to 5 of 50 tickets`. This is not correct, there are 100 tickets in the database. The query returned only 50 of them, so that would confuse the user.

Reply:
> I wrote my answers below so you know my logic, but I think I might not understand. I think you're right, but it would be nice to set up a call for you to explain it to me. I think the best thing would be to set up a database with like 150 records and see how the table handles it
> 
> > Why exactly do you want 100 records shown all at once? when they get around 50-75 service requests per week?
> 
> I don't think 100 records should be shown at once, just that the SQL query should ask for a higher number, like 100. The frontend table would only display 10 or so at a time
> 
> > Can I ask why you think this pagination is limited to only two pages, since from your comment it is not a question/wonder - it is a statement?
> 
> My understanding might be wrong, but this was my logic. The SQL query here returns the latest records, but up to a limit of 10. The pages that you are displaying on the table are of size 5. If the query returns at most 10 records and the front end table displays 5 records at a time, then the max number of pages could be 10/5 = 2
> 
> > No, it is not limited to two pages only- with the current limit set, if there are 100 records passed on the table it will be 10 pages holding 10 records on each page. limit(int) refers to how many rows we want displayed on each page
> 
> Oh I think I must have a misunderstanding then! So will this query go through the whole database using SQL?
> 
> > These screenshots were taken on my local machine, I created 7 records manually because I have to create them manually... If there is a fast way to create 100s of records in minutes I will do so and if I knew about it I would have done it so but sadly I do not.
> 
> I could help with that! We could use a query with a loop to do that. I could help you write that
> 
> > Maybe we should think about creating testing seed data?
> 
> Great idea!
> 
> > From the meeting, they said they get 50- 75 requests a week. If they are closing these tickets they will be moving out of this table and on the closet tickets section which we will only be rendering I think Kate said 20 recently closed tickets so if we use this same table still does not make sense to flood their dashboard of 3-4 tables showing 100 of records each.
> 
> Definitely don't want massive tables, I just want to make sure we can handle a large dataset when we use this tool for different use cases.
> 
> > Supabase has built pagination on the server side [here](https://www.restack.io/docs/supabase-knowledge-supabase-pagination-guide) if they get 50 to 75 the most weekly it will take a few months for their db to have 1000 records. You can add a few lines of code to the existing queries to leverage the pagination feature built-in.
> 
> Oh great let me check that out

**SORRY**, I actually was thinking about a different limit when I comment on these. ( I really should stop doing that before I head to bed LOL).  I was thinking of page rows lol.  So I want to apologize for my misunderstanding about pages and yes I agree the limit none nonexistent at this point. I think we originally limited it because that query was designed for the recent tickets, and this was even prior to the concept of pagination because initially, we did not think they would not have "recent tickets" passing 10 or so per day I think also this was design was also tied to the rush demo tied in with view tickets logic.

- But I do stand by what I said 100 records per table shown at once is too much hence I think why we thought about the pagination solution.

- we should remove the limit since we want all of the service request tickets and then use our filters and use ui pagination to control the view of the user and also do what @jnakaso  above what @sharrief said and your original thought about service side pagination even it would take them months to get there might as well future proof it.

Reply:
> @tristan-88 is right we don't expect too many records, but we do need to build a solution that will grow with SH. This function is specifically for loading recent tickets on the dashboard. We will have other pages that show the entire client list, or pet list, or service details list. When we get to those pages, we will likely have different fields to show which means writing a different query, at which point we can use limit and offset, as Tristian linked to in the Supabase docs.
> 
> I don't think we're looking to have pagination for the Recent Tickets list as it appears on the dashboard. The task, per my understanding, is to build re-usable client components. So the pagination Tristian has on the client will work for this use case. A future table on a "all clients" page will need to implement its query and data fetching to take server pagination into consideration.
> 
> Just my opinion. What do you think @DillonOLeary ?

Agreed on future proofing, yah we can utilized these on the clients and pets pages as well.